pkg/git/client.go

```diff
@@ -0,0 +1,197 @@
+package git
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/cli/safeexec"
+)
+
+var remoteRE = regexp.MustCompile(`(.+)\s+(.+)\s+\((push|fetch)\)`)
+
+type Client struct {
+	command  func(ctx context.Context, args ...string) (*exec.Cmd, error)
+	dir      string
+	exeErr   error
+	exePath  string
+	stderr   io.Writer
+	stdin    io.Reader
+	stdout   io.Writer
+	workTree string
+}
+
+type ClientOptions struct {
+	Command  func(ctx context.Context, args ...string) (*exec.Cmd, error)
```

Comment:
I don't love exposing something that is necessary for testing purposes only but couldn't think of a better way to go about this.